server/console.go

```diff
@@ -235,15 +222,52 @@ func consoleInterceptorFunc(logger *zap.Logger, config Config) func(context.Cont
 			return nil, status.Error(codes.Unauthenticated, "Console authentication required.")
 		}
 
-		if !checkAuth(config, auth[0]) {
+		if ctx, ok = checkAuth(ctx, config, auth[0]); !ok {
 			return nil, status.Error(codes.Unauthenticated, "Console authentication invalid.")
 		}
-
+		role := ctx.Value(ctxConsoleRoleKey{}).(console.UserRole)
+		forbidden := false
+		switch role {
```

Comment:
would be good to have 4 `const maps[string]interface{}{{"/nakama.console.Console/AddUser": nil},...}` types, and avoid the nested switch statement